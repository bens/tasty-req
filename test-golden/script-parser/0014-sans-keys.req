1: POST /
{"x": [{"a": 0, "b": 1, "c": 2}]}
---
{:r 1?["x"][0] / "a" "b"} <> {"b": {:r 1?["a"]}}
