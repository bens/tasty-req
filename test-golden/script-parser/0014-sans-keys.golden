[ Command
    { command'id = 1
    , command'always = False
    , command'method = "POST"
    , command'url = [ Right "/" ]
    , command'request_body =
        Just
          (JsonObject
             (Object
                (fromList
                   [ ( "x"
                     , JsonArray
                         [ JsonObject
                             (Object
                                (fromList
                                   [ ( "a" , JsonInteger 0 )
                                   , ( "b" , JsonInteger 1 )
                                   , ( "c" , JsonInteger 2 )
                                   ]))
                         ]
                     )
                   ])))
    , command'response_body =
        Just
          (JsonCombine
             (JsonCustom
                "r"
                (RespRef
                   (Ref 1 Request [ Right "x" , Left 0 ] (fromList [ "a" , "b" ]))) :|
                [ JsonObject
                    (Object
                       (fromList
                          [ ( "b"
                            , JsonCustom
                                "r" (RespRef (Ref 1 Request [ Right "a" ] (fromList [])))
                            )
                          ]))
                ]))
    }
]
